test/e2e/reconciler/clusterworkspacedeletion/controller_test.go

```diff
@@ -227,19 +227,23 @@ func TestWorkspaceDeletionController(t *testing.T) {
 				t.Logf("Ensure workspace in the org workspace is deleted")
 				require.Eventually(t, func() bool {
 					wslist, err := server.orgKcpClient.TenancyV1alpha1().ClusterWorkspaces().List(ctx, metav1.ListOptions{})
+					// 404 could be returned if the org workspace is deleted.
```

Comment:
Why isn't this also true for the namespace list calls above this?

Reply:
I believe this is coming from https://github.com/kcp-dev/kcp/blob/decced493792aeda1e171b9c860cc9efbcff00fd/pkg/server/apiextensions.go#L147. I see this getting logged repeatedly when the e2e is about to fail, presumably from the test trying to list clusterworkspaces every 1s

Reply:
Hm, so you don't get a 404 when you ask for a built-in type even when the workspace does not exist? Interesting.

Reply:
```
k get --raw '/clusters/root:bob/api/v1/namespaces'
{"kind":"NamespaceList","apiVersion":"v1","metadata":{"resourceVersion":"1983"},"items":[]}
```

Reply:
What I don't like here: with this change we do not test what we wanted to test anymore. I think we have to use this etcd client pattern of checking directly in etcd. A 404 of a list does not say anything about deletion of the objects.

Reply:
https://github.com/kcp-dev/kcp/issues/992

Reply:
Let's get this in, then follow up